tasks/README.md

```diff
@@ -107,19 +112,39 @@ Any parameter that does not specify a type will accept any value (default type i
 
 If a parameter type accepts `null` the task runner will accept either a `null` value or the absence of the property. Task authors must accept missing properties for nullable parameters. Task authors must not differentiate between absent properties and properties with `null` values.
 
+### Options
+
+**type**: A Puppet Type string describing the type of value the parameter accepts. Default is `Any`. (rev 1)
+
+**description**: A string description of the parameter. (rev 1)
+
+## Task implementations
+
+The implementations property is used to describe different implementations for differet targets. The value is an array of implementation options objects. For each target the task is run on the Task Runner must compare the required features of the target to it's list of available features on that target and execute the first implementation where all requirements are statisfied.
+
+### Options
+
+**name**: The file name of the task file that contains the implementation of the task. This must be at the top level of the tasks directory of the module.(rev 2)
+
+**requirements**: A list of features the target is required to support for the implementation to be used. `shell`, `powershell` and `puppet-agent` features are added by the ssh, winrm, and pcp transports respectively. (rev 2)
+
+**input_method**: The input method to use for this implementation of the task. Default empty [] which will make this implementation suitable for all targets. (rev 3)
+
+**files**: files required by this implementation. These will be concatenated with the files array from the top level of the tasks, metadata. (rev 3)
+
 ### Metaparameters
 
 In addition to the tasks parameters, the task runner may inject metaparameters prefixed by `_`.
 
-**_noop**: Used to implement logic in tasks to support cases where task should not perform certain actions.
+**_noop**: Used to implement logic in tasks to support cases where task should not perform certain actions. (rev 1)
 
-**_task**: Allow multiple task implementations to access the same executable file. The `_task` metaparameter provides the executable the task name to allow task specific logic to be implemented. 
+**_task**: Allow multiple task implementations to access the same executable file. The `_task` metaparameter provides the executable the task name to allow task specific logic to be implemented. (rev 3)
```

Comment:
The only distinction I've noticed between rev 2 and rev 3 is that we claim Bolt supports rev 2. So should we put `_task` and `_installdir` as rev 2?

Reply:
_task at least. Bolt doesn't yet support installdir right?

Reply:
Ah, right `_installdir` is not in yet.